src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -1053,12 +1055,20 @@ def __init__(self,
 
         self.events: List[OrchestratorEvent] = events or []
 
+        self.virtual_ip = virtual_ip
+        self.ports = ports
+
     def service_type(self) -> str:
         return self.spec.service_type
 
     def __repr__(self) -> str:
         return f"<ServiceDescription of {self.spec.one_line_str()}>"
 
+    def get_port_summary(self) -> str:
+        if not self.ports:
+            return ''
+        return f"{self.virtual_ip or '?'}:{','.join(map(str, self.ports or []))}"
```

Comment:
in case of no virtual ip, should we add the `host addr` here? cc @varshar16

Reply:
At this level, I think no.  ``orch ps`` output might include an IP for a specific daemon instance, but at the service level, we can/should only show an IP here if there is a stable (virtual) IP for the service that is independent of daemon (re)placement, scheduling, etc.